pkg/guid/guid.go

```diff
@@ -0,0 +1,47 @@
+package guid
+
+import (
+	"fmt"
+	"net"
+)
+
+// GUID address is an net.HardwareAddr encapsulation
+type GUID net.HardwareAddr
+
+// ParseGUID parses string only as GUID 64 bit
+func ParseGUID(s string) (GUID, error) {
+	ha, err := net.ParseMAC(s)
+	if err != nil {
+		return nil, err
+	}
+	if len(ha) != 8 {
+		return nil, fmt.Errorf("invalid GUID address %s", s)
+	}
+
+	return GUID(ha), nil
+}
+
+// String returns the string representation of GUID
+func (g GUID) String() string {
+	return net.HardwareAddr(g).String()
+}
+
+func (g GUID) Uint64() uint64 {
+	return uint64(g[0])<<56 | uint64(g[1])<<48 | uint64(g[2])<<40 | uint64(g[3])<<32 |
+		uint64(g[4])<<24 | uint64(g[5])<<16 | uint64(g[6])<<8 | uint64(g[7])
+}
+
+// nextGUID return next guid
+func (g GUID) nextGUID() GUID {
```

Comment:
why it not a private method in the guildpool?

Reply:
moved